### PR TITLE
Testing - ResultsItemDetails

### LINKED ISSUE #108: Unit test ResultsItemDetails
- [x] All elements render as expected
- [x] The `savePark` method is fired when the save button is clicked

### PR DESCRIPTION
**Pull request by nicolegooden**
### Participating Group Members:

- @nicolegooden 

### Code Highlights:

- Set up required store, route, computed, etc. for test suite
- Added unit test to ensure all expected elements render correctly
- Added unit test to ensure the `savePark` method is called once when the save button is clicked

### Have Tests Been Added?

- [ ] No.
- [X] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

I have skipped one test because it is a draft with some comments.  Would love your feedback on testing mutations from the Vuex store if anyone has any insight.

### Any background context you want to provide?

N/A

### Message/Questions for reviewer:

N/A

### Issues:

- Closes #108 

### Tracking Consistency:

- [X] added appropriate labels
- [X] My code follows the code style of this project and has removed all unnecessary annotations
- [ ] I have added comments on my pull request, particularly in hard-to-understand areas
- [ ] All new and existing tests passed

- [X] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: tests/unit/ResultsItemDetails.spec.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils';
+import ResultsItemDetails from '@/components/ResultsItemDetails.vue';
+import Vuex from 'vuex';
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+describe('ResultsItemDetails', () => {
+  let state, store, mutations, computed, $route;
+
```
Comment (stacyp2006) [inline tests/unit/ResultsItemDetails.spec.js:10]: Nice! I always forget that you can do multiple declarations in one line. So fresh and so clean.

File: tests/unit/ResultsItemDetails.spec.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    expect(getAllPTags.at(0).text()).toBe('Address: 5123-5275 Valmont Rd, Boulder');
+    expect(getAllPTags.at(1).text()).toBe('Closed Now');
+    expect(getAllPTags.at(2).text()).toBe('Rating: 4.7 / 5');
+
+    const getAllButtonTags = wrapper.findAll('button');
+    expect(getAllButtonTags.at(0).text()).toBe('SAVE');
+    expect(getAllButtonTags.at(1).text()).toBe('Get Directions');
+    expect(getAllButtonTags.at(2).text()).toBe('Search Again');
+  })
+
```
Comment (stacyp2006) [inline tests/unit/ResultsItemDetails.spec.js:108]: Big expect section here, but nicely organized.

**Comment by bretm9** [comment]
Thanks for getting these tests in! I'll have to look into testing mutations, but great job figuring out how to test methods!

**Review by bretm9 — APPROVED** [review]
Thanks for getting these tests in! I'll have to look into testing mutations, but great job figuring out how to test methods!

File: tests/unit/ResultsItemDetails.spec.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    }) 
+    const saveParkButton = wrapper.find('.button-save-park');
+    saveParkButton.trigger('click');
+    expect(store._mutations.savePark).toHaveBeenCalled();
+    //This test is currently failing.
+    //I believe this is due to the fact that the savePark method
+    //is not being mocked directly.  I am wondering if it is not
+    //possible to test the mutation this way, and that it needs to 
+    //be an isolated test...
+  })
```
Comment (stacyp2006) [inline tests/unit/ResultsItemDetails.spec.js:153]: Maybe you've already see this, but possibly some good info here: https://vue-test-utils.vuejs.org/v2/guide/vuex.html#testing-with-a-mock-store

**Review by stacyp2006 — APPROVED** [review]
Testing looks good! Hope the link I shared is helpful w/Vuex mutation testing.

**Review by npdarrington — APPROVED** [review]
Overall looks very good and clean! Looking forward to seeing the mutation assertion work when we have the chance to come back to it.

**Pull request merged**